src/main/java/frc/robot/util/Vision.java

```diff
@@ -0,0 +1,176 @@
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.estimator.DifferentialDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+import frc.robot.constants.FieldConstants;
+import frc.robot.constants.VisionConstants;
+
+
+
+public class Vision {
+  private static RobotPoseEstimator robotPoseEstimator;
+  // private static RobotPoseEstimator robotPoseEstimator2;
+  private static AprilTagFieldLayout aprilTagFieldLayout;
+ 
+
+  /**
+   * Calls the other setup function
+   */
+  public static void setup() {
+    setup(new PhotonCamera(VisionConstants.kCameraName1), new PhotonCamera(VisionConstants.kCameraName2));
+  }
+
+  /**
+   * Sets up the cameras and pose estimator
+   * @param camera1 The first camera
+   * @param camera2 The second camera
+   */
+  public static void setup(PhotonCamera camera1, PhotonCamera camera2) {
+    ArrayList<Pair<PhotonCamera, Transform3d>> camList;
+    if(VisionConstants.k2Cameras){
+      camList = new ArrayList<Pair<PhotonCamera, Transform3d>>(List.of(
+        new Pair<PhotonCamera, Transform3d>(camera1, VisionConstants.kRobotToCamera1),
+        new Pair<PhotonCamera, Transform3d>(camera2, VisionConstants.kRobotToCamera2)
+      ));
+    }else{
+      camList = new ArrayList<Pair<PhotonCamera, Transform3d>>(List.of(
+        new Pair<PhotonCamera, Transform3d>(camera1, VisionConstants.kRobotToCamera1)
+      ));
+    }
+    getTagFieldLayout();
+    aprilTagFieldLayout.setOrigin(AprilTagFieldLayout.OriginPosition.kBlueAllianceWallRightSide);
+    
+    robotPoseEstimator = new RobotPoseEstimator(aprilTagFieldLayout, PoseStrategy.CLOSEST_TO_REFERENCE_POSE, camList);
+
+    if(VisionConstants.k2Cameras){
+      // ArrayList<Pair<PhotonCamera, Transform3d>>camList = new ArrayList<Pair<PhotonCamera, Transform3d>>(List.of(
+      //   new Pair<PhotonCamera, Transform3d>(camera2, VisionConstants.kCameraToRobot2)
+      // ));
+      // robotPoseEstimator2 = new RobotPoseEstimator(aprilTagFieldLayout, PoseStrategy.AVERAGE_BEST_TARGETS, camList);    
+    }
+
+  }
+
+  public static AprilTagFieldLayout getTagFieldLayout() {
+    // AprilTagFields.loadAprilTagLayoutField();
+    try {
+      //TODO: Fix this
+      // aprilTagFieldLayout = AprilTagFieldLayout.loadFromResource(AprilTagFields.k2023ChargedUp.m_resourceFile);
+      aprilTagFieldLayout = AprilTagFieldLayout.loadFromResource(AprilTagFields.k2022RapidReact.m_resourceFile);
```

Comment:
I saw your writing that this doesn't work, why doesn't it? Is the file not there yet?

Reply:
The April Tag locations in the file are for RapidReact (i.e., for testing on last year's field). Same for tag locations in kDefault.

Reply:
I put the AprilTag locations into a new pull request. The April Tag locations are part of the PowerPlay field.

Reply:
I don't know why. It can't find k2023ChargedUp. I think the file needs to be updated, but I can't find it anywhere.